Jenkinsfile

```diff
@@ -170,8 +176,11 @@ pipeline {
             withGithubNotify(context: 'Unit Tests', tab: 'tests') {
               deleteDir()
               unstash 'source'
-              dir("${BASE_DIR}"){
-                sh(label: 'Run Unit tests', script: './script/jenkins/unit-test.sh')
+              dir(BASE_DIR){
+                retry(2) { // Retry in case there are any errors to avoid temporary glitches
+                  sleep randomNumber(min: 5, max: 10)
+                  sh(label: 'Run Unit tests', script: './script/jenkins/unit-test.sh')
+                }
```

Comment:
I suggest to only add the retry for the `linux build` and `windows build` for now, as those were the stages where I mainly saw issues, and automatically retrying e.g. system tests would increase build times quite a bit in case real failures are found.

Reply:
That's indeed a consequence of using this approach, sure thing, I'll then keep this strategy for those builds.